### PR TITLE
Adjusted various UI elements `border-radius`

### PR DESCRIPTION
**Pull request by jaredpdesigns**
This PR increases the overall "roundness" of many elements in the UI, which rely on the `rounded` class from Tailwind. Primary changes made:
- Removed the overwrite of `DEFAULT` value for `border-radius` inside of `tailwind.preset.js` which reset the `rounded` class value to equal 4px.
- Adjusted a few elements which had behavior different than using `rounded` including `InputGroup`
- Increased the radius to 8px for `Card` and `Tab` using the `rounded-lg` class.
- Added roundness to `Table` which may need to reverted based on feedback.

> Part of https://github.com/hypothesis/frontend-shared/issues/1304

### CODE REVIEW

**Comment by robertknight** [comment]
A few screenshots of what this branch looks like in the client:

Annotation cards:

Rounded corners

With these changes we now have three distinct levels of rounded-ness in the UI for the top bar and menus (square), left sidebar buttons (small rounding) and annotation cards (large rounding). We probably are going to need different levels of roundedness for different elements, but we should have a coherent story for when to use each.

Tabs. Note there is an issue with the connection between the tab list and the panels:

Tabs

Dialog:

Annotation card

Menus:

These are currently still square.

Menu roundedness

**Comment by robertknight** [comment]
LMS assignment configuration front page:

LMS assignment details

There is an issue here with the corners which we'll need to fix.

LMS assignment picker URL entry dialog:

LMS enter URL

LMS Canvas files picker:

LMS canvas files picker

LMS YouTube video picker:

LMS YouTube video picker

**Comment by jaredpdesigns** [comment]
@robertknight just a few comments:
- Regarding tabs, the best way to adjust this is `<Card classes={selectedTab === 'shared' ? 'rounded-tl-none' : undefined}>` where the class to remove rounding from the adjacent `Card` is conditional, I had implemented that in one of the demo pages in the Pattern Library.
- In the case of some of the LMS examples, specifically the Assignment Configuration page, adding an additional class of `overflow-hidden` is a heavy-handed way to confirm the roundness of the parent element masks any children.
- Menus would be another good thing to review, is this an element in the Pattern Library? In general agree with the 3 levels of roundness. In my opinion most everything that isn't a card should defer to using the `rounded` class, which defaults to 4px, and only element that are either the Card itself or "touching" elements like the tabs should take on the `rounded-lg` class which doubles the roundness to 8px.

**Comment by acelaya** [comment]
> Menus would be another good thing to review, is this an element in the Pattern Library?

No, menus are currently defined in `client`. See [Menu](https://github.com/hypothesis/client/blob/38f8caa2331cca213d083c04b1b21e580add4866/src/sidebar/components/Menu.tsx), [MenuSection](https://github.com/hypothesis/client/blob/38f8caa2331cca213d083c04b1b21e580add4866/src/sidebar/components/MenuSection.tsx) and [MenuItem](https://github.com/hypothesis/client/blob/38f8caa2331cca213d083c04b1b21e580add4866/src/sidebar/components/MenuItem.tsx)

**Comment by jaredpdesigns** [comment]
@acelaya I'd be happy to spin up a separate PR to look at stuff in the client that the Pattern Library doesn't cover. Looks like we are using Tailwind here as well. However, if it’s loading Tailwind via the `frontend-shared` repo we may need to release the changed there first then look at the client.

**Comment by acelaya** [comment]
> However, if it’s loading Tailwind via the frontend-shared repo we may need to release the changes there first then look at the client.

We usually use `yalc` to test unreleased frontend-shared changes in the other projects.

You can see the instructions here https://github.com/hypothesis/frontend-shared/blob/f3ac2d1a3c7d804e0b6c77c42e472286dbab1c4b/docs/developing.md#testing-locally-with-other-projects

**Comment by jaredpdesigns** [comment]
@robertknight @acelaya is there more here that needs to be done before considering merging it? Not sure what is remaining to consider. Does it just need some other kind of approval?

cc: @dwhly

**Comment by acelaya** [comment]
Hey @jaredpdesigns.

The main consideration before merging this, is that the way it is now, it's a breaking change. That means it would block adoption of this library, and make it harder to release other feature or fixes here.

That's why we suggested testing it in every consuming project with `yalc`, before merging it.

Another option (probably preferable, now that I think about it) would be to try and make it non-breaking, by keeping the `borderRadius` configuration as it is, and letting downstream projects adopt rounded corners and overwrite that config at their own pace.

That way we can use this PR as a sandbox just to make every component here ready for rounded corners (the other changes you did), but without introducing side effects if downstream needs to update to newer frontend-shared versions for other unrelated reasons.

On a later stage, once every project has adopted rounded corners, we can remove the `borderRadius` option from the tailwind preset here.

There are a few grey areas though, which are the components moving here from `rounded` to `rounded-lg` and such, but I think that, considering the rounded corner size we have now, it is negligible. Worth double checking though.

**Comment by robertknight** [comment]
I agree with @acelaya that it would probably be best for us to roll out the changes one project at a time, and then fold them back into the pattern library when we are done. We could do as @acelaya suggested and roll out the change in each individual project's config first, and then later fold it back into the pattern library. Or we could change the defaults here in a new major version and override them downstream until we are happy that the revised styles are working as desired.

I think it would be quite useful if we could feature flag style changes like this. Alejandro and I can talk about it and see how much effort that would require. Perhaps we could do something with CSS variables for example.

**Comment by acelaya** [comment]
> Is there anything else I need to do at this point?

I think you are good 🙂 

During this sprint we'll do a couple more checks and then merge this with the needed adjustments.

**Comment by acelaya** [comment]
I have continued testing this, and found a couple of edge cases that needed to be fixed. Having in mind the our roll-out plan, these are the things I have changed :

> I have pushed the changes in additional commits.

- `CardHeader` with `variant="secondary"` was rendering a background color "outside" the card.
  - I have fixed it by adding `rounded-t-[inherit]` to the `CardHeader`.
- I had mixed feelings about adding rounded corners to tables, and I have found some problems with that.
  1. Sticky headers were no longer working due to the `overflow-hidden` added to `table`.
  2. Tables inside scrollable containers looked odd, due to rounded corners being only visible when at the very top or very bottom.
  
  [not-working.webm](https://github.com/hypothesis/frontend-shared/assets/2719332/cbae2e3a-4403-4a38-8174-fd1c29149f2d)
  - My proposed solution is to not make tables have rounded corners by default, and instead wrap them in a container with rounded corners when needed.
  - This required adding a new `borderless` property to tables to make them look good when wrapped in a container which renders its own border.
  
  [working.webm](https://github.com/hypothesis/frontend-shared/assets/2719332/4dce108a-2d7e-4b05-a46c-31ce6c5ac388)
- I noticed our `Scroll` did not have rounded corners.
  - I kept this as the default behavior, but added a new `rounded` prop to `ScrollBox` and `ScrollContainer` that allows for scrollable containers to render rounded corners that are always visible.
  This is useful for cases like the above (tables) but can be useful in other cases.
- As discussed in previous comments, the changes in tailwind preset are not backward compatible.
  - I have reverted back to what we had before, with two changes:
    1. We not only define `DEFAULT`, but also `lg`, because I have noticed it was too obvious and still a breaking change.
    2. I have defined the config in a way that downstream projects can overwrite it via CSS vars, allowing for this to be dynamically enabled at runtime (for example with feature flags).
    ```
    borderRadius: {
      lg: 'var(--h-border-radius-lg, 0.125rem)',
      DEFAULT: 'var(--h-border-radius, 0.125rem)',
    }
    ```
    3. Next major version can drop this completely, but all downstream projects should update first.

###  TBD

There's a few things that still need testing before merging this PR.

- I would like Cards to automatically drop the top-left rounded corner when used together with tabs and the first tab is selected. Let's see if I can manage to get this done.
- These changes still need further testing with LMS, via and client. I'll be doing that today.

File: src/components/data/ScrollBox.tsx
Comment (acelaya) [inline src/components/data/ScrollBox.tsx:10]: It's hard to imagine the different scenarios in which we use scrollable containers, so I'm adding rounded corners as an optional opt-in feature

File: src/tailwind.preset.js
Comment (acelaya) [inline src/tailwind.preset.js:27]: Downstream projects should not see any changes when updating the library, but they can overwrite the rounded border values via `--h-border-radius` and `--h-border-radius-lg` vars.

We will need to find a way to define CSS vars based on feature flags.

File: styles/library.scss
Comment (acelaya) [inline styles/library.scss:10]: I have enabled rounded corners in the pattern library.

**Comment by acelaya** [comment]
Tested the changes in client. It looks good, and there's no changes in the UI.

I also verified we can add something like this in our styles:

```css
html[data-rounded-corners] {
  --h-border-radius: 0.25rem;
  --h-border-radius-lg: 0.5rem;
}
```

And then use a feature flag to conditionally add the `data-rounded-corners` attribute to the html node. Something in those lines should allow us to use a feature flag to enable the feature conditionally.

**Comment by jaredpdesigns** [comment]
@acelaya this is all sounding really good. I did run into the issue with tables, because the outermost element is the only convenient way to target sub-children which aren't rounded by default, wrapping it in a rounded parent sounds like a most effective solution. Great work!

File: src/components/data/ScrollBox.tsx
Comment (robertknight) [reply src/components/data/ScrollBox.tsx:10]: To confirm, the current default of `false` matches the existing look?

File: src/pattern-library/components/patterns/data/TablePage.tsx
Comment (robertknight) [inline src/pattern-library/components/patterns/data/TablePage.tsx:160]: What was the reason for picking this example to add rounding to - because it was already customizing the scroll container?

File: src/pattern-library/components/patterns/data/TablePage.tsx
Comment (robertknight) [inline src/pattern-library/components/patterns/data/TablePage.tsx:288]:
```suggestion
                outer borders, in case it is rendered in a container with its
```

File: src/pattern-library/components/patterns/prototype/TabbedDialogPage.tsx
Comment (robertknight) [inline src/pattern-library/components/patterns/prototype/TabbedDialogPage.tsx:145]: Unrelated to the changes here, this component is called "TabbedDialogPage" but that doesn't match the Import/Export label in the sidebar. We also might want to extract this pattern as a component in future to make it easier to instantiate?

**Review by robertknight — APPROVED** [review]
LGTM as far as I can see in the pattern library.

File: src/components/data/ScrollBox.tsx
Comment (acelaya) [reply src/components/data/ScrollBox.tsx:10]: Yes. It will continue being non-rounded (current behavior) unless actively opting in.

File: src/pattern-library/components/patterns/data/TablePage.tsx
Comment (acelaya) [reply src/pattern-library/components/patterns/data/TablePage.tsx:160]: No particular reason. Just to avoid having so many very similar examples.

I'm open to keep this one as is and add an extra example with a rounded scroll.

File: src/pattern-library/components/patterns/prototype/TabbedDialogPage.tsx
Comment (acelaya) [reply src/pattern-library/components/patterns/prototype/TabbedDialogPage.tsx:145]: > We also might want to extract this pattern as a component in future to make it easier to instantiate?

Agree.

It would also make it easier to transparently handle the rounded corners when the first tab is selected, so that you don't need to remember every time this pattern is used.

**Pull request merged**
